browser/ui/tabs/test/brave_tab_strip_model_browsertest.cc

```diff
@@ -206,3 +233,33 @@ IN_PROC_BROWSER_TEST_F(BraveTabStripModelRenamingTabBrowserTest,
   EXPECT_EQ(1u, pending_commands.size());
   EXPECT_EQ(/*kCommandAddTabExtraData*/ 33, pending_commands[0]->id());
 }
+
+IN_PROC_BROWSER_TEST_F(BraveTabStripModelRenamingTabBrowserTest,
+                       CustomTitleShouldBeResetWhenOriginChanges) {
+  BraveTabStripModel* tab_strip_model =
+      static_cast<BraveTabStripModel*>(browser()->tab_strip_model());
+  ASSERT_FALSE(
+      TabRendererData::FromTabInModel(tab_strip_model, 0).is_custom_title);
+  auto* web_contents = tab_strip_model->GetWebContentsAt(0);
+  ASSERT_EQ(web_contents->GetLastCommittedURL(), "about:blank");
+  ASSERT_FALSE(
+      TabRendererData::FromTabInModel(tab_strip_model, 0).is_custom_title);
+
+  tab_strip_model->SetCustomTitleForTab(0, u"Custom Title");
+  ASSERT_EQ(TabRendererData::FromTabInModel(tab_strip_model, 0).title,
+            u"Custom Title");
+  ASSERT_TRUE(
+      TabRendererData::FromTabInModel(tab_strip_model, 0).is_custom_title);
+
+  // When navigating to a different origin, the custom title should be reset
+  // to the default title.
```

Comment:
maybe also add a test that a page-initiated navigation (like via `document.location.href = 'https://example.com'`) triggers the change?